### PR TITLE
Fix tags not being applied to restic snapshots

### LINKED ISSUE #619: Tags are not assigned to snaspshots
### Description

It seems like Tags are not working currently for Backup and Schedule Resources.

K8up Helm Version: 2.0.2

### Additional Context

_No response_

### Logs

```shell
restic -r s3://s3.eu-central-1.amazonaws.com/$bucket


repository b5944e6c opened successfully, password is correct
ID        Time                 Host         Tags        Paths
---------------------------------------------------------------------------
cdf9bb5b  2022-03-10 15:29:16  default              /data/foo

```

### Expected Behavior

```
repository b5944e6c opened successfully, password is correct
ID        Time                 Host         Tags        Paths
---------------------------------------------------------------------------
cdf9bb5b  2022-03-10 15:29:16  default     foobar     /data/foo
---------------------------------------------------------------------------
```

### Steps To Reproduce

```
cat <<EOF | kubectl apply -f -
apiVersion: k8up.io/v1
kind: Backup
metadata:
  name: backup-test
spec:
  keepJobs: 4
  tags:
  - foobar
  backend:
    repoPasswordSecretRef:
      name: backup-s3-secret
      key: RESTIC_PASSWORD
    s3:
      endpoint: s3.eu-central-1.amazonaws.com
      bucket: backup-k8s-staging
      accessKeyIDSecretRef:
        name: backup-s3-secret
        key: AWS_ACCESS_KEY_ID
      secretAccessKeySecretRef:
        name: backup-s3-secret
        key: AWS_SECRET_ACCESS_KEY
EOF
```

### Version of K8up

ghcr.io/k8up-io/k8up:v2.1.0

### Version of Kubernetes

v1.18.20

### Distribution of Kubernetes

Rancher

### PR DESCRIPTION
**Pull request by Kidswiss**
## Summary

Closes #619

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`,
      as they show up in the changelog
- [x] Update the documentation.
- [x] Update tests.
- [x] Link this PR to related issues.

### CODE REVIEW

File: config/samples/deployments/pvc-example.yaml [hunk 1]
```diff
@@ -1,15 +1,13 @@
 kind: PersistentVolumeClaim
 apiVersion: v1
 metadata:
-  name: myclaim
   annotations:
     k8up.io/backup: 'true'
   namespace: myproject
```
Comment (ccremer) [inline config/samples/deployments/pvc-example.yaml:6]: Is that correct? Do you intent to remove the `name` and not `namespace`?

File: e2e/test-06-tagged-backup.bats [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# shellcheck disable=SC2034
+DETIK_CLIENT_NAME="kubectl"
+# shellcheck disable=SC2034
+DETIK_CLIENT_NAMESPACE="k8up-e2e-subject"
+# shellcheck disable=SC2034
+DEBUG_DETIK="true"
+
+@test "Given a PVC, When creating a Backup with tags, Then expect Restic snapshot with tags" {
+	expected_content="expected content: $(timestamp)"
+	expected_filename="expected_filename.txt"
```
Comment (ccremer) [inline e2e/test-06-tagged-backup.bats:16]: is it really worth having and maintaining an e2e test for this (presumably rather rare) use case?
Is an integration test or even just unit test not enough to verify that the container contains some args?

Each e2e test case adds 1-2 minutes of pipeline run

File: config/samples/deployments/pvc-example.yaml (hunk 1 above)
Comment (Kidswiss) [reply config/samples/deployments/pvc-example.yaml:6]: Yeah that's wrong...

File: e2e/test-06-tagged-backup.bats (hunk 2 above)
Comment (Kidswiss) [reply e2e/test-06-tagged-backup.bats]: I've switched to an integration test.

File: controllers/backup_it_utils_test.go [hunk 3]
```diff
@@ -285,3 +289,21 @@ func (ts *BackupTestSuite) notifyObserverOfBackupJobStatusChange(status k8upObse
...
+func (ts *BackupTestSuite) assertJobHasTagArguments(job *batchv1.Job) {
+	JobArguments := job.Spec.Template.Spec.Containers[0].Args
+	found := false
+	for i := range JobArguments {
+		if JobArguments[i] == backupTag {
+			found = true
+			break
+		}
+	}
+	ts.Assert().True(found, "did not find tag in job argmuments")
```
Comment (ccremer) [inline controllers/backup_it_utils_test.go:308]: I think there's a `ts.Assert().Contains` function that you can use to check the existence of an element in a string slice.
```suggestion
	jobArguments := job.Spec.Template.Spec.Containers[0].Args
	ts.Assert().Contains(jobArguments, backupTag, "backup tag in job args")
```

File: controllers/backup_it_utils_test.go [hunk 4]
```diff
@@ -285,3 +289,14 @@ func (ts *BackupTestSuite) notifyObserverOfBackupJobStatusChange(status k8upObse
 	event.Event = status
 	observer.GetUpdateChannel() <- event
 }
+
+func (ts *BackupTestSuite) newBackupWithTags() *k8upv1.Backup {
+	BackupWithTags := ts.newBackup()
+	BackupWithTags.Spec.Tags = []string{backupTag}
+	return BackupWithTags
```
Comment (ccremer) [inline controllers/backup_it_utils_test.go:296]: variable names should be camelCase
```suggestion
	backupWithTags := ts.newBackup()
	backupWithTags.Spec.Tags = []string{backupTag}
	return backupWithTags
```

**Review by ccremer — APPROVED** [review]

**Pull request merged**